master/internal/api_workspace.go

```diff
@@ -273,13 +287,9 @@ func (a *apiServer) PostWorkspace(
 		return nil, status.Error(codes.PermissionDenied, err.Error())
 	}
 
-	if len(req.Name) < 1 {
-		return nil, status.Errorf(codes.InvalidArgument,
-			"name '%s' must be at least 1 character long", req.Name)
-	}
-	if len(req.Name) > 80 {
-		return nil, status.Errorf(codes.InvalidArgument,
-			"name '%s' must be at most 80 character long", req.Name)
+	err = validateWorkspaceName(req.Name)
+	if err != nil {
```

Comment:
minor formatting suggestion, I believe this can be condensed into one line 

```
if err = validateWorkspaceName(req.Name); err != nil {
     return nil, err
}
```